File: tests/adversarial/sql-injection-cs-001/nested.cs
using Microsoft.EntityFrameworkCore;
using System.Linq;

namespace App {
    public class UserRepository {
        public void GetUser(MyContext context, string name) {
            void query() {
                var users = context.Users.FromSqlRaw($"SELECT * FROM users WHERE Name = '{name}'").ToList();
            }
            query();
        }
    }
}

File: tests/adversarial/sql-injection-rs-001/nested.rs
use sqlx::sqlite::SqlitePool;

pub async fn get_user(pool: &SqlitePool, name: &str) {
    async fn inner(p: &SqlitePool, n: &str) {
        let users = sqlx::query(&format!("SELECT * FROM users WHERE name = '{}'", n))
            .fetch_all(p)
            .await
            .unwrap();
    }
    inner(pool, name).await;
}
